apps/ui/src/components/RevisionsAlert.tsx

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2025 The Ontario Institute for Cancer Research. All rights reserved
+ *
+ * This program and the accompanying materials are made available under the terms of
+ * the GNU Affero General Public License v3.0. You should have received a copy of the
+ * GNU Affero General Public License along with this program.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import { pcglColours } from '@/providers/ThemeProvider';
+import { CaretRightFilled, ExclamationCircleFilled } from '@ant-design/icons';
+import { GeneralType, RevisionType } from '@pcgl-daco/validation';
+import { Collapse, Flex, theme, type CollapseProps } from 'antd';
+import { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+
+interface RevisionsProps {
+	sectionRevisions: RevisionType;
+	general?: GeneralType;
+}
+
+const RevisionsAlert = ({ sectionRevisions, general }: RevisionsProps) => {
+	const { token } = theme.useToken();
+
+	const { t: translate } = useTranslation();
+	const [filterRevisions, setFilteredRevisions] = useState<CollapseProps['items'] | undefined>(undefined);
+	const [generalComments, setGeneralComments] = useState<CollapseProps['items'] | undefined>(undefined);
+
+	useEffect(() => {
```

Comment:
This is a long useEffect, could you add some comments at the top to explain what is being done in this effect and when it is meant to trigger?

Can some of this be separated out into functions that are defined outside the useEffect?